src/Type/InspectorTypeExtension.php

```diff
@@ -107,6 +107,16 @@ private function specifyAssertAll(MethodReflection $staticMethodReflection, Stat
             return new SpecifiedTypes();
         }
 
+        $traversable = $node->getArgs()[1]->value;
+        $traversableInfo = $scope->getType($traversable);
+
+        // If it is already not mixed (narrowed by other code, like
+        // '::assertAllArray()'), we could not provide any additional
+        // information. We can only narrow this method to 'array<mixed, mixed>'.
+        if (!$traversableInfo instanceof MixedType) {
```

Comment:
```suggestion
        if (!$traversableInfo->equals(new MixedType())) {
```

Can we use `equals` to avoid calls of `instanceof`? Lots of code in phpstan-drupal needs to be fixed, but PHPStan has slowly been moving away from performing these kind of checks on types over using its type methods

Reply:
Yes, it works the same :+1: